[supervisor] nil deref in parseInstanceConfigs

Will not start. I captured a download and recovered my pending changes. I will leave this workspace alone for you to inspect if you wish. Let me know if you do not need it.

So, this was a fun dive. But it yielded a nil deref in the end:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xa4ce27]

goroutine 99 [running]:
github.com/gitpod-io/gitpod/supervisor/pkg/ports.parseInstanceConfigs(0xc00000e0c0, 0x1, 0x1, 0xc0007b0480, 0xc00064a988, 0xc0000aff20, 0xc00064aa38)
	/tmp/build/components-supervisor--app.ab31968bf49cbf8ddfd44ce3fd31fb8dffa29d82/_deps/gitpod-core-components-supervisor--app/pkg/ports/ports-config.go:184 +0xa7
github.com/gitpod-io/gitpod/supervisor/pkg/ports.(*ConfigService).update(0xc0003824e0, 0xc0001823f0, 0xc0000afee8, 0x2)
	/tmp/build/components-supervisor--app.ab31968bf49cbf8ddfd44ce3fd31fb8dffa29d82/_deps/gitpod-core-components-supervisor--app/pkg/ports/ports-config.go:159 +0x7d
github.com/gitpod-io/gitpod/supervisor/pkg/ports.(*ConfigService).Observe.func1(0xc0005340c0, 0xc000600120, 0xc0003824e0, 0xdbea20, 0xc000520000)
	/tmp/build/components-supervisor--app.ab31968bf49cbf8ddfd44ce3fd31fb8dffa29d82/_deps/gitpod-core-components-supervisor--app/pkg/ports/ports-config.go:138 +0x2f2
created by github.com/gitpod-io/gitpod/supervisor/pkg/ports.(*ConfigService).Observe
	/tmp/build/components-supervisor--app.ab31968bf49cbf8ddfd44ce3fd31fb8dffa29d82/_deps/gitpod-core-components-supervisor--app/pkg/ports/ports-config.go:112 +0xac
```

--- a/components/supervisor/pkg/ports/ports-config.go
+++ b/components/supervisor/pkg/ports/ports-config.go
@@ -181,6 +181,10 @@ func parseWorkspaceConfigs(ports []*gitpod.PortConfig) (portConfigs map[uint32]*
 
 func parseInstanceConfigs(ports []*gitpod.PortsItems) (portConfigs map[uint32]*gitpod.PortConfig, rangeConfigs []*RangeConfig) {
 	for _, config := range ports {
+		if config == nil {
+			continue
+		}
+
 		rawPort := fmt.Sprintf("%v", config.Port)
 		Port, err := strconv.Atoi(rawPort)
 		if err == nil {

